keras/callbacks.py

```diff
@@ -671,19 +673,25 @@ def set_model(self, model):
         else:
             self.writer = tf.summary.FileWriter(self.log_dir)
 
-        self.saver = tf.train.Saver()
-        self.embeddings = {layer.name: (layer.weights[0],
-                                        os.path.join(self.log_dir, layer.name +
-                                                     '.ckpt'))
-                           for layer in self.model.layers
-                           if layer.name in self.embeddings_layer_names}
-        self.embeddings_metadata_setup()
+        if self.embeddings_freq:
+            self.saver = tf.train.Saver()
+
+            embeddings_layer_names = self.embeddings_layer_names \
+                                     if self.embeddings_layer_names else \
+                                        [layer.name for layer in self.model.layers
+                                         if type(layer).__name__ == 'Embedding']
+            self.embeddings = {layer.name:
```

Comment:
It's possible to write this one liner in several more readable lines of code. It will also solve your pep8 problems.